### DISCUSSION TITLE
The difference of ADShaftConnectedPump1PhaseUserObject and ADPump1PhaseUserObject

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#25750] [Q&A Modules: Thermal Hydraulics]
Hi all,

I understand the process and target of ADShaftConnectedPump1PhaseUserObject and ADPump1PhaseUserObject, but I'm really confused about the momentum and energy residual terms in computeFluxesAndResiduals function:
1. why do the momentum and energy source residual terms in ADPump1PhaseUserObject need to multiply 0.5 but these terms in ADShaftConnectedPump1PhaseUserObject don't?
2. why the energy source term in  ADShaftConnectedPump1PhaseUserObject does not include the pump head term, just like in ADPump1PhaseUserObject?

In ADPump1PhaseUserObject:
![image](https://github.com/idaholab/moose/assets/118275988/0e174c4f-1e92-450a-9e36-3323cbcb705a)

In ADShaftConnectedPump1PhaseUserObject 
![image](https://github.com/idaholab/moose/assets/118275988/699bd0ab-ac69-43b4-84d9-e172a2884180)

Could anyone kindly tell me? 
Thank you very much!

**Comment by lindsayad** [comment]
@joshuahansel @licharlot

**Comment by joshuahansel** [comment]
Hi,

1. The shaft-connected one is enclosed in an `if (c == 0)` block, so it only gets added for the first connection, so this is the same as adding that term twice with the 0.5 factor in the other UO. Confusing.
2. The shaft-connected pump is just having an entirely different formulation for the energy term because it has all of the information from the pump curves. The other pump just makes a very generic approximation of this term because it doesn't have anything else. The former is preferred if you have the curves.

**Reply by SomnusYu** [reply]
I got it. Thank you very much!

**Reply by SomnusYu** [reply]
By the way, can you show me the figures of head and torque data used in the shaft_connected_pump_1phase teat file? I can not find the data figure in the offering references:
Semiscale Program, summarized in NUREG/CR-4945.
LOFT Systems Tests, described in NUREG/CR-0247.

**Reply by joshuahansel** [reply]
@licharlot do you know where they might have come from? If not in the references, then I don't know.

**Reply by licharlot** [reply]
The conventional homologous curves are in the references,  page 177 of https://www.osti.gov/servlets/purl/6302855 (NUREG/CR-4945) and page 161 of https://www.nrc.gov/docs/ML0702/ML070250396.pdf (NUREG/CR-0247). 

They must have been used as a basis to develop the polar homologous curves that are used by this component. The process is described in the component documentation.

**Reply by SomnusYu** [reply]
Thank you very much!!

**Reply by SomnusYu** [reply]
Hi @joshuahansel and @licharlot, 

I still have a question: why does the homologous_torque need to multiply -1, which is different from the description in component documentation?

documentation:
![image](https://github.com/idaholab/moose/assets/118275988/abcc6401-1b10-4da3-a74b-cc98d10b65d2)
![image](https://github.com/idaholab/moose/assets/118275988/83d4fd9d-57dd-4a06-ac34-0f4c91e1cd3d)

code:
![image](https://github.com/idaholab/moose/assets/118275988/0b771ed8-5151-465a-a99a-c4b29485fded)

**Reply by licharlot** [reply]
This is simply a convention. The hydraulic torque if later on multiplied by -1 to be used as a source in the energy equation, and is negative to be applied on the shaft speed equation.

**Reply by SomnusYu** [reply]
I understand. Thank you！
